configure

```diff
@@ -341,10 +352,89 @@ cscope="${CSCOPE-cscope}"
 
 confighdefs=''
 
+# define options for debugging
+if ${debug}
+then
+    cflags="${CFLAGS--pedantic -MMD -Wall -Wextra -Og -fno-inline -ggdb}"
+else
+    defconfigh "NDEBUG"
+fi
+
+
+# check if the compiler works
+checking 'whether the compiler works'
+cat >"${tempsrc}" <<END
+int main(void) { return 0; }
+END
+if
+    trymake
+then
+    checked "yes"
+elif
+    [ x"${cc}" = x"${_host}c99" ] && [ x"${CC+set}" != x"set" ] && (
+        cc="${_host}gcc"
+        trymake
+    )
+then
+    cc="${_host}gcc"
+    checked "yes (using $cc)"
+elif
+    [ x"${cc}" = x"${_host}c99" ] && [ x"${CC+set}" != x"set" ] && (
+        cc="${_host}clang"
+        trymake
+    )
+then
+    cc="${_host}clang"
+    checked "yes (using $cc)"
+else
+    checked "no"
+    printf 'Compiler "%s" not found or not working.\n' "${CC-${cc}}" >&2
+    printf 'The compiler is expected to accept these options: %s\n' \
+        "${CFLAGS-${cflags}} ${CADDS-${null}}" >&2
+    fail
+fi
+
+# check if we are cross compiling
+checking 'whether we are cross compiling'
+if
+    tryexec
+then
+    checked "no"
+    cross_compiling=false
+else
+    checked "yes"
+    cross_compiling=true
+fi
 
 # check OS type
 checking 'operating system'
-ostype=$(uname -s | tr "[:upper:]" "[:lower:]")
+if [ -n "$OS" ]; then
+    ostype="$OS"
+elif [ -n "$cross_compiling" ]; then
```

Comment:
Since 86121d3a4c4d7f2c3e680b89e5091962faf2ec39, `$cross_compiling` expands to `true` or `false`.

```suggestion
elif $cross_compiling; then
```